core/src/main/scala/org/http4s/Uri.scala

```diff
@@ -317,6 +320,37 @@ object Uri {
       val merge = segments.initLast.map(_._1).getOrElse(segments)
       Path(merge.concat(path.segments), absolute = absolute, endsWithSlash = path.endsWithSlash)
     }
+
+    def startsWith(path: Path): Boolean = {
+      val i = segments.iterator
+      val j = path.segments.iterator
+      while (j.hasNext && i.hasNext) if (i.next() != j.next())
+        return false
+
+      !j.hasNext
+    }
+
+    def startsWith_(path: String): Boolean = startsWith(Path.fromString(path))
```

Comment:
Underscore to avoid overloading?

Reply:
yes

Reply:
I like the impulse, but I think of the underscore suffix as being related to `Unit` (e.g., `traverse_`).  Would `startsWithString` be too verbose?